src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,95 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 1 << 4;
+    private static final double LOAD_FACTOR = 0.75;
+    private Node<K, V>[] table;
+    private int size;
+    private int capacity;
+
+    public MyHashMap() {
+        this.capacity = DEFAULT_INITIAL_CAPACITY;
+        this.table = (Node<K, V>[]) new Node[capacity];
+    }
 
     @Override
     public void put(K key, V value) {
+        resize();
+        int pos = getIndexFromHash(hash(key));
+        if (table[pos] == null) {
+            table[pos] = new Node<>(key, value);
+            size++;
+        } else if (Objects.equals(key, table[pos].key)) {
+            table[pos].value = value;
```

Comment:
no need to handle this specifically, this will be handled by that while loop

Reply:
while loop starts its work by checking table[pos].next , otherwise it doesnt work :(

Reply:
Because this check "if (Objects.equals(key, bucketNode.key))" should be done first in the loop, like in getValue method, then it will not fail